src/main/java/agh/ics/oop/SimulationEngine.java

```diff
@@ -0,0 +1,49 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SimulationEngine implements IEngine{
+
+    private final MoveDirection[] moveDirections;
+    private final IWorldMap map;
+    private final Vector2d[] startPositions;
```

Comment:
to raczej nie musi być atrybut - korzystamy tylko w konstruktorze, który może przekazać parametr bezpośrednio do `addAnimalToMap()`